src/ad.jl

```diff
@@ -23,22 +23,18 @@ function scalar_nlsolve_ad(prob, alg, args...; kwargs...)
     return sol, partials
```

Comment:
@ChrisRackauckas, what does this code (mathematically) exactly do? I have never messed around with ForwardDiff so not really sure.

The code doesn't work if $u$ is an SVector. The previous tests had a typo and never tested that case.

Reply:
It's handling the derivative rule for nonlinear solve, just skipping the dual numbers in the nonlinear solve and then applying the partials at the solution point.

Reply:
Done!